osquery/tables/system/darwin/smc_keys.cpp

```diff
@@ -379,28 +382,99 @@ inline uint32_t strtoul(const char *str, size_t size, size_t base) {
   return total;
 }
 
-float getConvertedValue(const std::string &smcType, const std::string &smcVal) {
-  // Convert hex string to decimal.
+inline uint64_t strtoull(const char* str, size_t size, size_t base) {
+  uint64_t total = 0;
+  for (size_t i = 0; i < size; i++) {
+    total += static_cast<uint64_t>(static_cast<unsigned char>(str[i]))
+             << (size - 1 - i) * 8;
+  }
+  return total;
+}
+
+double getConvertedValue(const std::string& smcType,
```

Comment:
Changed this (and all related vars) from `float` to `double` to handle 64-bit types.  We don't utilize any in tables right now, but there are definitely some 64-bit SMC keys (18 on my system) and this future-proofs us.